src/services/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from '@angular/router';
+
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { AuthenticationService } from './authentication.service';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class AuthGuard implements CanActivate {
+
+    constructor(
+        private router: Router,
+        private authenticationService: AuthenticationService,
+    ) { }
+
+
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+        return this.authenticationService.checkAuthentication()
+            .pipe(
+                map((res) => {
```

Comment:
Could `find` be used here? https://www.learnrxjs.io/operators/filtering/find.html

Reply:
not really searching through for a value though? but if you have a code example of how this would look please share, however as a simple "if...else" I don't think it's needed?

Reply:
The only example I found was this, but I not sure its better:

```typescript
const somethings$ = source$
  .filter(isSomething)
  .do(something):

const differentThings$ = source$
  .filter(!isSomething)
  .do(aDifferentThing):

// merge them together
const onlyTheRightThings$ = somethings$
  .merge(differentThings$)
  .do(correctThings)
```